src/index.ts

```diff
@@ -1,155 +1,19 @@
-// indexAuth.js
-// BASE SETUP
-// =============================================================================
-
-import bodyParser from 'body-parser'
-import { asArray, asObject } from 'cleaners'
-import cluster from 'cluster'
-import cors from 'cors'
-import {
-  autoReplication,
-  forkChildren,
-  makePeriodicTask
-} from 'edge-server-tools'
-import express from 'express'
-import http from 'http'
-import nano from 'nano'
-import promisify from 'promisify-node'
-
 import { config } from './config'
-import { asExchangeRateReq, getExchangeRate, ReturnRate } from './rates'
-// const REQUIRED_CODES = ['BC1', 'DASH', 'LTC', 'BCH']
-
-const asExchangeRatesReq = asObject({
-  data: asArray(asExchangeRateReq)
+import { exchangeRateRouter } from './exchangeRateRouter'
+import { createRouter } from './router'
+import { createServer } from './server'
+import { logger } from './utils/utils'
+
+// Create Router
+const router = createRouter({
+  // Create Exchange Rate Router
+  '/v1': exchangeRateRouter(config)
 })
-
-const AUTOREPLICATION_DELAY = 1000 * 60 * 30 // 30 minutes
-const EXCHANGE_RATES_BATCH_LIMIT = 100
-
-// call the packages we need
-const app = express()
-
-const mylog = console.log
-
-function dateString(): string {
-  const date = new Date()
-  return date.toDateString() + ':' + date.toTimeString()
-}
-
-// configure app to use bodyParser()
-// this will let us get the data from a POST
-app.use(bodyParser.urlencoded({ extended: true }))
-app.use(bodyParser.json({ limit: '50mb' }))
-app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }))
-app.use(cors())
-
-// Nano for CouchDB
-// =============================================================================
-const nanoDb = nano(config.couchUri)
-const dbRates = nanoDb.db.use('db_rates')
-promisify(dbRates)
-
-// ROUTES FOR OUR API
-// =============================================================================
-const router = express.Router()
-
-// middleware to use for all requests
-router.use(function(req, res, next) {
-  // do logging
-
-  mylog('Something is happening.')
-  next() // make sure we go to the next routes and don't stop here
+// Create Server
```

Comment:
Use the cluster/initDb convention as seen in edge-info-server. Refer to edge-server-tools